manim/utils/bezier.py

```diff
@@ -369,3 +370,108 @@ def get_quadratic_approximation_of_cubic(a0, h0, h1, a1):
 
 def is_closed(points: typing.Tuple[np.ndarray, np.ndarray]) -> bool:
     return np.allclose(points[0], points[-1])
+
+
+def proportions_along_bezier_curve_for_point(
+    point: typing.Iterable[typing.Union[float, int]],
+    control_points: typing.Iterable[typing.Iterable[typing.Union[float, int]]],
+    round_to: typing.Optional[typing.Union[float, int]] = 1e-6,
+) -> np.ndarray:
+    """Obtains the proportion along the bezier curve corresponding to a given point
+    given the bezier curve's control points.
+
+    Parameters
+    ----------
+    point
+        The Cartesian Coordinates of the point whose parameter
+        should be obtained.
+    control_points
+        The Cartesian Coordinates of the ordered control
+        points of the bezier curve on which the point may
+        or may not lie.
+    round_to
+        A float whose number of decimal places all values
+        such as coordinates of points will be rounded.
+
+    Returns
+    -------
+        np.ndarray[float]
+            List containing possible parameters (the proportions along the bezier curve)
+            for the given point on the given bezier curve.
+            This usually only contains one or zero elements, but if the
+            point is, say, at the beginning/end of a closed loop, may return
+            a list with more than 1 value, corresponding to the beginning and
+            end etc. of the loop.
+
+    Raises
+    ------
+    :class:`ValueError`
+        When ``point`` and the control points have different shapes.
+    """
+    if not all(np.shape(point) == np.shape(c_p) for c_p in control_points):
+        raise ValueError(
+            f"Point {point} and Control Points {control_points} have different shapes."
+        )
+
+    control_points = np.array(control_points)
+    n = len(control_points) - 1
+
+    roots = []
+    for dim, coord in enumerate(point):
+        control_coords = control_points[:, dim]
+        terms = []
+        for term_power in range(n, -1, -1):
+            outercoeff = choose(n, term_power)
+            term = []
+            sign = 1
+            for subterm_num in range(term_power, -1, -1):
+                innercoeff = choose(term_power, subterm_num) * sign
+                subterm = innercoeff * control_coords[subterm_num]
+                if term_power == 0:
+                    subterm -= coord
+                term.append(subterm)
+                sign *= -1
+            terms.append(outercoeff * sum(np.array(term)))
+
+        bezier_polynom = np.polynomial.Polynomial(terms[::-1])
+        polynom_roots = bezier_polynom.roots()
+        if len(polynom_roots) > 0:
+            roots.append(polynom_roots)
```

Comment:
Do we really want to filter away empty root arrays here? I think that's what gives false positives sometimes. Imagine if one dimension finds roots and the second doesn't. If that empty list is skipped, `roots` becomes just [[some_root]] and the intersect call will treat it as a common root, because no list does *not* have it.

Also, sometimes roots contains arrays of differing lengths, which causes the `np.around` call below to error out. An example of that is
`proportions_along_bezier_curve_for_point([0, 1], [[-1,1],[0,0], [1,1]])`
which gives `roots = [array([0.5]), array([0., 1.])]`

Reply:
Yep, this was the problem. Both issues are now fixed. `np.around` is now called for every polynomial's root separately, and empty root sets are also considered when `np.intersect1d`-ing.